agent.py

```diff
@@ -445,8 +445,13 @@ def _submit_jmx_service_discovery(self, jmx_sd_configs):
             return
 
         if self.supervisor_proxy is not None:
-            jmx_state = self.supervisor_proxy.supervisor.getProcessInfo(JMX_SUPERVISOR_ENTRY)
-            log.debug("Current JMX check state: %s", jmx_state['statename'])
+            try:
+                jmx_state = self.supervisor_proxy.supervisor.getProcessInfo(JMX_SUPERVISOR_ENTRY)
+                log.debug("Current JMX check state: %s", jmx_state['statename'])
+                return
+            except Exception as e:
+                log.exception("Unable to get JMXFetch process state from supervisor: %s", e)
```

Comment:
Could you also fit this: _"cannot submit JMX auto-discovery configurations"_  into the message somehow?